rfc/0058-error-handling.md

```diff
@@ -317,6 +317,37 @@ A Note on IDs: The IDs in this RFC are only for organisational purposes and MUST
 
 * Raised when decoding of the data into the user object failed
 
+### 21 RateLimitingFailure
```

Comment:
Regarding whether to have a single RateLimitingFailure with multiple causes, or separate errors: for me RateLimitedMaxConnections is a bit of an odd man out and should be its own error.  The other three indicate a temporary rate limit that can be resolved by just trying again later - IMO an application would probably handle all 3 the same way, so no need for a separate error there.  For RateLimitedMaxConnections it's more serious and probably going to require some manual resolution, such as running fewer apps or reducing number of KV endpoints.